ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/peertask/PeerTaskExecutor.java

```diff
@@ -98,8 +101,8 @@ public <T> PeerTaskExecutorResult<T> execute(final PeerTask<T> peerTask) {
       if (peer.isEmpty()) {
         executorResult =
             new PeerTaskExecutorResult<>(
-                Optional.empty(), PeerTaskExecutorResponseCode.NO_PEER_AVAILABLE);
-        continue;
+                Optional.empty(), PeerTaskExecutorResponseCode.NO_PEER_AVAILABLE, Optional.empty());
+        break;
```

Comment:
As it was, it was immediately querying EthPeers again, resulting in the same behavior. This might be a good place to use my new WaitForPeerTask replacement (I'll put a PR up soon)

Reply:
I agree with that, retrying immediately has no benefit